tiff-2-png-converter.py

```diff
@@ -1,197 +1,358 @@
 import argparse
 import sys
-import os 
+import os
 import cv2
 import numpy as np
 import imagesize
 import math
-import tifffile  as tiff
+import tifffile as tiff
 import numpngw
-from progress_bar import *
-
-verbose = False
-
-
-
-
-def convert_tiff_directory_action(source_path, dest_path):
-    list_files,width, heigth = list_folder_files(source_path)
-    num_images = len(list_files)
-    image_r = [None] * num_images
-    image_g = [None] * num_images
-    image_b = [None] * num_images
-    images = [None] * num_images
-    bits_per_sample = ""
-    printProgressBar(0, num_images, prefix = 'grouping channels:', suffix = 'Complete', length = 50)
-    for i in range(num_images):
-        file_full_path = list_files[i]
-        img = tiff.imread(list_files[i])
-        if bits_per_sample=="":
-            bits_per_sample = img.dtype
-        elif bits_per_sample!=img.dtype:
-            print("image " + file_full_path + "has a different sample type "+bits_per_sample+" expected: "+bits_per_sample)
-            sys.exit("Check images bits per sample tag")
-        name, extension = os.path.splitext(file_full_path)
-        if "ch1" in name:
-             if verbose:
-                print("Load Image " + str(i) + " Channel 1 - "+ file_full_path )
-             image_r[i] = img
-        elif "ch2" in name:
-             if verbose:
-                print("Load Image " + str(i) + " Channel 2 - "+ file_full_path )
-             image_g[i] = img
-        elif "ch3" in name:
-             if verbose:
-                print("Load Image " + str(i) + " Channel 3 - "+ file_full_path )
-             image_b[i] = img
+from progress_bar import print_progress_bar
+from image_process_helper import *
+from scipy.interpolate import interp1d
+import re
+
+import typer
+
+
+class TiffProcessor:
+    verbose = False
+    equalize_histogram = False
+    equalize_histogram_method = None
+    gamma = False
+    gamma_val = 1
+    num_imgs = None
+    source = None
+    dest = None
+
+    @staticmethod
+    def add_arguments(parser):
+        parser.add_argument(
+            "-s",
+            "--source",
+            required=True,
+            help="Path to directory where the image sequence is located",
+        )
+        parser.add_argument(
+            "-d",
+            "--dest",
+            required=True,
+            help="Path to the directory and filename of the result merged image (png by default) will be saved. ",
```

Comment:
could you put an example here? It's unclear to me if the filename is required or not

Reply:
It is specify it that the arg is required.
let me rework this using typer.